File: Minimum Number of Pushes to Type Word II.py
class Solution:
    def minimumPushes(self, word: str) -> int:
        counter = Counter(list(word))
        q = [-val for val in counter.values()]
        q.sort()
        pushes = 0
        num_pad = 0
        rotation = 1
        for count in q:
            pushes += -count * rotation
            num_pad += 1
            if num_pad == 8:
                rotation += 1
                num_pad = 0
        return pushes

File: Minimum Number of Pushes to Type Word II.cpp
class Solution {
public:
    int minimumPushes(string word) {
        map<char, int> counter;
        for (char c: word) counter[c]++;
        vector<int> q;
        for (auto [_, v]: counter) q.push_back(-v);
        sort(q.begin(), q.end());
        int pushes = 0, num_pad = 0, rotation = 1;
        for (int count : q) {
            pushes += -count * rotation;
            num_pad++;
            if (num_pad == 8) {
                rotation++;
                num_pad = 0;
            }
        }
        return pushes;
    }
};
